test/controllers/tags_controller_test.rb

```diff
@@ -170,8 +170,61 @@ class TagsControllerTest < ActionController::TestCase
     sign_in users(:deleter)
     patch :update, params: { id: categories(:main).id, tag_id: tags(:topic).id,
                              tag: { parent_id: tags(:child).id, excerpt: 'things' } }
-    assert_response 400
+    assert_response :bad_request
     assert_not_nil assigns(:tag)
     assert_equal ["The #{tags(:child).name} tag is already a child of this tag."], assigns(:tag).errors.full_messages
   end
+
+  test 'should correctly rename a tag' do
+    sign_in users(:admin)
```

Comment:
Should we test these as `:moderator` rather than `:admin`?  Moderators *should* be able to do this; if we ever had a bug in that, admin super-privileges might mask it.